vaadin-text-field-flow-parent/vaadin-text-field-flow/src/test/java/com/vaadin/flow/component/textfield/binder/AbstractTextFieldValidationTest.java

```diff
@@ -0,0 +1,130 @@
+package com.vaadin.flow.component.textfield.binder;
+
+import com.vaadin.flow.component.HasValue;
+import com.vaadin.flow.data.binder.Binder;
+import com.vaadin.flow.data.binder.BindingValidationStatus;
+import com.vaadin.flow.data.binder.BindingValidationStatusHandler;
+import com.vaadin.flow.function.SerializablePredicate;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.*;
+
+public abstract class AbstractTextFieldValidationTest<T> {
+
+    private static final String BINDER_FAIL_MESSAGE = "BINDER_VALIDATION_FAIL";
+    private static final String REQUIRED_MESSAGE = "REQUIRED";
+
+    public static class Bean<T> {
+        private T property;
+
+        public T getProperty() {
+            return property;
+        }
+
+        public void setProperty(T property) {
+            this.property = property;
+        }
+    }
+
+    protected abstract HasValue<?, T> getField();
+
+    protected abstract void setValidValue();
+
+    protected abstract void setComponentInvalidValue();
+
+    protected abstract void setBinderInvalidValue();
+
+    protected abstract void setEmptyValue();
+
+    protected abstract SerializablePredicate<? super T> getValidator();
+
+    @Captor
+    private ArgumentCaptor<BindingValidationStatus<?>> statusCaptor;
+
+    @Mock
+    private BindingValidationStatusHandler statusMock;
+
+    @Before
+    public void init() {
+        MockitoAnnotations.openMocks(this);
+    }
```

Comment:
**suggestion:** 

If we instantiate the field in the `init` method:

```suggestion
    public void init() {
        MockitoAnnotations.openMocks(this);
        
        field = getField();
    }
```

...we can simplify `getField` in subclasses:

```diff
protected HasValue<?, BigDecimal> getField() {
-  if (field == null) {
-      field = new BigDecimalField();
-  }
-  return field;
+  return new BigDecimalField();
}
```

Reply:
Somewhat applied the proposed change. Let me know what you think.